ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/tabsontop/v1/tabsontop/clientlibs/site/js/tabs.js

```diff
@@ -326,15 +339,200 @@
          * @private
          * @param {Number} index The index of the item to navigate to
          */
-        #navigateAndFocusTab(index) {
+        #navigateAndFocusTab(tabId) {
             var exActive = this._active;
-            this.#navigate(index);
-            this.focusWithoutScroll(this._elements["tab"][index]);
+            this.#navigate(tabId);
+            this.focusWithoutScroll(this.#getTabNavElementById(tabId));
+        }
+
+        handleChildAddition(childView){
+            if(childView.getInstanceManager()!=null && this._templateHTML[childView.getInstanceManager().getId()]!=null){
+                var navigationTabToBeRepeated = this._templateHTML[childView.getInstanceManager().getId()]['navigationTab'].cloneNode(true);
+                navigationTabToBeRepeated.id = childView.id+"__tab";
+                navigationTabToBeRepeated.setAttribute("aria-controls",childView.id+"__tabpanel");
+                var instanceIndex = childView.getModel().index;
+                var instanceManagerId = childView.getInstanceManager().getId()
+                if(instanceIndex==0){
+                    var closestNonRepeatableFieldId=this._templateHTML[instanceManagerId]['closestNonRepeatableFieldId'];
+                    var closestRepeatableFieldInstanceManagerId=this._templateHTML[instanceManagerId]['closestRepeatableFieldInstanceManagerId'];
+                    var indexToInsert=this.#getTabIndexToInsert(closestNonRepeatableFieldId,closestRepeatableFieldInstanceManagerId);
+                    if(indexToInsert==0) {
+                        var tabListParentElement = this.#getTabListElement();
+                        tabListParentElement.insertBefore(navigationTabToBeRepeated, tabListParentElement.firstChild);
+                    }else{
+                        var beforeElement = this.#getCachedTabs()[indexToInsert-1];
+                        beforeElement.after(navigationTabToBeRepeated);
+                    }
+                }else {
+                    var beforeTabNavElementId = childView.getInstanceManager().children[instanceIndex - 1].element.id + "__tab"
+                    var beforeElement = this.#getTabNavElementById(beforeTabNavElementId);
+                    beforeElement.after(navigationTabToBeRepeated);
+                }
+                this.#cacheElements(this._elements.self);
+                var repeatedTabPanel = this.#getTabPanelElementById(childView.id+"__tabpanel");
+                repeatedTabPanel.setAttribute("aria-labelledby",childView.id+"__tab");
```

Comment:
we should have a `idSuffix` prop at view level to avoid hard coding this everytime.

Reply:
added them, thanks